### PR TITLE
#48258 Automate smoke tests for Book Store module

### PR DESCRIPTION
**Pull request by lorenacioata**
The following TCs are automated:

C157 - Check if Add to Your Collection button is enabled when logged in
C162 - Check if Add to You Collection button is not visible when not logged in
C155 - Check if click on Go to Book Store button Profile page changes to Book Store page

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "@wdio/spec-reporter": "^7.10.1",
     "chromedriver": "^92.0.2",
     "prettier": "^2.3.2",
-    "wdio-chromedriver-service": "^7.2.0"
+    "protractor": "^7.0.0"
```
Comment (TaiaDimitrova) [inline package.json:33]: Why do you need this package?

File: test/pageobjects/BookStore.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+class BookStore {
+  get bookStoreBtn() {
+    return $("#item-2*=Book Store");
+  }
+  get profileBtn() {
+    return $("#item-3*=Profile");
+  }
+  get bookGitGuide() {
+    return $("=Git Pocket Guide");
```
Comment (TaiaDimitrova) [inline test/pageobjects/BookStore.js:9]: Please, try to use css selectors, attribute selectors, instead searching by text, cause text might change more often that the selector of an element.

File: test/pageobjects/BookStore.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  }
+  get addToCollectionBtn() {
+    return $$("#addNewRecordButton")[1];
```
Comment (TaiaDimitrova) [inline test/pageobjects/BookStore.js:12]: Is this really working in terms of using async method?

File: tests/book-store-smoke.spec.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    await expect(await addToCollection).is.equal(true);
+  });
+  it("Check if Add to You Collection button is not visible when not logged in", async () => {
+    await App.openBookStorePage();
+    if (await LoginLogout.logoutButton.isDisplayed()) {
+      await (await LoginLogout.logoutButton).click();
+      await BookStore.accessBookStore();
+      await BookStore.accessBookGitGuide();
+      await expect(
+        await BookStore.addToCollectionBtn.isDisplayedInViewport()
```
Comment (TaiaDimitrova) [inline tests/book-store-smoke.spec.js:23]: I'd recommend to use hooks for logging in. Current approach is duplicating the login steps.

**Review by TaiaDimitrova — CHANGES_REQUESTED** [review]
Please take into consideration review comments from the PR #10 .

File: test/pageobjects/BookStore.js (hunk 3 above)
Comment (lorenacioata) [reply test/pageobjects/BookStore.js:12]: What do you mean? Can you be more specific please?

File: tests/HomePage.spec.js [hunk 5]
```diff
@@ -1,69 +1,66 @@
...
-    const bookStoreCard = await (await HomePage.bookStoreCard).getText();
-    await expect(await bookStoreCard).contains("Book Store Application");
+    const cards = [
+      "Elements",
+      "Forms",
+      "Alerts, Frame & Windows",
+      "Widgets",
+      "Interactions",
+      "Book Store Application",
+    ];
```
Comment (TaiaDimitrova) [inline tests/HomePage.spec.js:25]: You have the cards defined in the objects/homePage.js file, import that object and use it here.

**Review by TaiaDimitrova — COMMENTED** [review]
I'd suggest to rename the objects folder to testData.

**Pull request merged**